pennylane/transforms/optimization/optimization_utils.py

```diff
@@ -110,49 +105,29 @@ def fuse_rot_angles(angles_1, angles_2):
         angles_1 (float): A set of three angles for the first ``qml.Rot`` operation.
         angles_2 (float): A set of three angles for the second ``qml.Rot`` operation.
 
-    Returns:
-        array[float]: A tuple of rotation angles for a single ``qml.Rot`` operation
-        that implements the same operation as the two sets of input angles.
+    Returns:s
+        array[float]: Rotation angles for a single ``qml.Rot`` operation that
+        implements the same operation as the two sets of input angles.
     """
-    # Check for all-zero instances; if there are some, we can just return the sum.
-    are_angles_1_zero = allclose(angles_1, zeros(3))
-    are_angles_2_zero = allclose(angles_2, zeros(3))
-
-    if are_angles_1_zero or are_angles_2_zero:
-        return stack([angles_1[i] + angles_2[i] for i in range(3)])
-
-    # RZ(a) RY(b) RZ(c) fused with RZ(d) RY(e) RZ(f)
-    # first produces RZ(a) RY(b) RZ(c+d) RY(e) RZ(f)
-    left_z = angles_1[0]
-    middle_yzy = angles_1[1], angles_1[2] + angles_2[0], angles_2[1]
-    right_z = angles_2[2]
-
-    # There are a few other cases to consider where things can be 0 and
-    # avoid having to use the quaternion conversion routine
-    # If b = 0, then we have RZ(a + c + d) RY(e) RZ(f)
-    if allclose(middle_yzy[0], 0.0):
-        # Then if e is close to zero, return a single rotation RZ(a + c + d + f)
-        if allclose(middle_yzy[2], 0.0):
-            return [left_z + middle_yzy[1] + right_z, 0.0, 0.0]
-        return stack([left_z + middle_yzy[1], middle_yzy[2], right_z])
-
-    # If c + d is close to 0, then we have the case RZ(a) RY(b + e) RZ(f)
-    if allclose(middle_yzy[1], 0.0):
-        # If b + e is 0, we have RZ(a + f)
-        if allclose(middle_yzy[0] + middle_yzy[2], 0.0):
-            return [left_z + right_z, 0.0, 0.0]
-        return stack([left_z, middle_yzy[0] + middle_yzy[2], right_z])
-
-    # If e is close to 0, then we have the case RZ(a) RY(b) RZ(c + d + f)
-    # The case where b is 0 actually already covered in the first loop,
-    # so only one case here.
-    if allclose(middle_yzy[2], 0.0):
-        return stack([left_z, middle_yzy[0], middle_yzy[1] + right_z])
-
-    # Otherwise, we need to turn the RY(b) RZ(c+d) RY(e) into something
-    # of the form RZ(u) RY(v) RZ(w)
-    u, v, w = _yzy_to_zyz(middle_yzy)
-
-    # Then we can combine to create
-    # RZ(a + u) RY(v) RZ(w + f)
-    return stack([left_z + u, v, w + right_z])
+
+    # Check if we are tracing; if so, use the special conditionals
+    if is_abstract(angles_1) or is_abstract(angles_2):
+        interface = _multi_dispatch([angles_1, angles_2])
```

Comment:
would this allow the 'removing zero rotations' and RZ gates to be used in the non-tracing cases in the previous file?